File: Code/Chapter 5/VB.NET/Exercise 5.3/Form1.vb
Imports System.Data
Imports System.Data.SqlClient
Imports System.Collections
Imports System.Data.Common

Public Class Form1
   Private dbRecordsHolder As ArrayList

   Private Sub btnPopulate_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnPopulate.Click
      Dim connectionString As String = "Server=(local);Database=Test;Integrated Security=SSPI"
      Using testConnection As SqlConnection = New SqlConnection(connectionString)
         Dim testCommand As SqlCommand = New SqlCommand("Select * from TestDemo", testConnection)
         testConnection.Open()
         Dim sqlDr As SqlDataReader = testCommand.ExecuteReader(CommandBehavior.CloseConnection)
         If sqlDr.HasRows Then
            Dim rec As DbDataRecord
            For Each rec In sqlDr
               dbRecordsHolder.Add(rec)
            Next
         End If
      End Using ' testConnection.Dispose is called automatically
   End Sub

   Private Sub btnDataBind_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnDataBind.Click
      myDataGrid.DataSource = dbRecordsHolder
   End Sub

   Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
      dbRecordsHolder = New ArrayList()
   End Sub
End Class

File: Code/Chapter 5/C#/Example 5.3/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

using System.Data.SqlClient ;
using System.Data.Common;
using System.Collections;

#endregion

namespace Example_5_3
{
    partial class Form1 : Form
    {
        private ArrayList dbRecordsHolder;
        public Form1()
        {
            InitializeComponent();
            dbRecordsHolder = new ArrayList();
        }

        private void btnPopulate_Click(object sender, EventArgs e)
        {
            string connectionString = "Server=(local);Database=Test;Integrated Security=SSPI";
            using (SqlConnection testConnection = new SqlConnection(connectionString))
            {
                SqlCommand testCommand =
                    new SqlCommand("Select * from TestDemo", testConnection);
                testConnection.Open();
                SqlDataReader sqlDr = testCommand.ExecuteReader(CommandBehavior.CloseConnection);

                if (sqlDr.HasRows)
                {
                    foreach (DbDataRecord rec in sqlDr)
                    {
                        dbRecordsHolder.Add(rec);
                    }
                }
            } // testConnection.Dispose is called automatically
        }

        private void btnDataBind_Click(object sender, EventArgs e)
        {
            myDataGrid.DataSource = dbRecordsHolder;
        }
    }
}